cookbook/bundles/tests.rst

```diff
@@ -0,0 +1,311 @@
+.. index::
+   single: Bundle; Testing third party bundles
+
+Writing Tests for Third Party Bundles
+=====================================
+
+The Standard Edition's front controller looks like this::
+
+    // web/app.php
+    use Symfony\Component\HttpFoundation\Request;
+
+    require_once __DIR__.'/../vendor/autoload.php';
+    require_once __DIR__.'/../app/AppKernel.php';
+
+    // Create the request from PHP variables super globals such as $_GET, $_POST, $_SERVER, etc
+    $request = Request::createFromGlobals();
+
+    // The application handles the Request
+    $kernel = new AppKernel('prod', false);
+    $response = $kernel->handle($request);
+
+    // Dump the Response's headers and print the Response's content
+    $response->send();
+
+In this example, Symfony is used as an application that mimicks the HTTP protocol,
+which means that it is possible to write transversal tests (System Tests) by building
+"manually" a ``Request``, giving it to our application and then check the returned ``Response``.
+
+This article provides examples on how to write such tests with third party bundles.
+
+Creating a Minimal AppKernel
+----------------------------
+
+Third party bundles cannot be ran standalone, they need an application.
+It is possible to create a minimalistic ``AppKernel`` inside the bundle for testing
+and showcase purposes, instead of:
+
+# Creating a separate application;
+# Install the bundle in it;
+# Then check how things turn out.
+
+If the third party bundle is named ``AcmeMyBundle``, the ``AppKernel`` could look like this::
+
+    // Tests/app/AppKernel.php
```

Comment:
I guess it's a matter of taste: some will argue that the tree grows unnecessarily deeper, others will agree that it keeps a "test application" separate from the tests.
To be fair, I used to put it in `Resources/app`, my point here is that there's many possibilities out there and each can be justified. This is a documentation article so nothing is forced on the developer, whatever we choose here won't prevent them to move it where they want. Maybe we should keep it this way and add a note?